crates/shared/src/order_validation.rs

```diff
@@ -647,17 +625,25 @@ impl OrderValidating for OrderValidator {
             },
         }
 
-        tracing::debug!(
-            ?uid,
-            ?order,
-            ?quote,
-            "checking if order is outside market price"
-        );
         // Check if we need to re-classify the market order if it is outside the market
         // price. We consider out-of-price orders as liquidity orders. See
         // <https://github.com/cowprotocol/services/pull/301>.
-        let class = match (class, &quote) {
-            (OrderClass::Market, Some(quote))
+        let (class, quote) = match class {
```

Comment:
This is getting quite hard to read. I'm thinking if we should maybe already get rid of the `liquidity` order type to simplify that whole logic quite a bit.
Then we could always `get_or_create_quote()` regardless and check if it's in market afterwards and do sth like:
```
fee.is_zero() && outside market => check limit order count
fee.is_zero() && inside market => all good; what we now consider a market order
fee.is_signed() => check if fee covers gas according to the quote
```

Reply:
I think exactly the same, it is like that in order to keep backwards compatibility. When do we plan to remove the `Liquidity` type? I could take care of that!